src/scripts/components/joinForm.js

```diff
@@ -55,45 +55,106 @@ class JoinForm {
+      // mobile is valid and email is valid - submit
+      // mobile is valid and email is invalid - error
+      // mobile is invalid and email is valid - error
+      // mobile is invalid and email is blank or invalid - error
+
+      // if mobile is blank and email exists
+      // email is valid - submit
+      // email is invalid - error
+
+      if (Mobile.value) {
+        if (_this.checkMobile(Mobile.value)) {
+          if (Email.value === '') {
+            _this.$suppressEmailSubscribe.val('1');
+            _this.$email.val(Mobile.value +'-smssubscriber@example.com');
+          }
+          _this.$smsTerms.val('sms_termsandconditions');
+          _this.$smsSubscribe.val('mobilesubscribe');
+          _this.$robodialTerms.val('yes');
+          
+          if (!_this.checkEmail(Email.value)) {
+            e.preventDefault();
+            Email.parentNode.classList.add("giraffe-has-errors");
+            _this.addErrorToList('Invalid email; please correct or remove');
+            return false;
+          }
+        } else /* mobile is invalid */ {
+          if (Email.value === '' || !_this.checkEmail(Email.value)) {
+            e.preventDefault();
+            Mobile.parentNode.classList.add("giraffe-has-errors");
+            Email.parentNode.classList.add("giraffe-has-errors");
+            _this.addErrorToList('Valid email OR mobile opt-in is required');
+            return false;
+          } else /* and email is valid */ {
+            e.preventDefault();
+            Mobile.parentNode.classList.add("giraffe-has-errors");
+            _this.addErrorToList('Invalid mobile; please correct or remove');
+            return false;
+          }
+        }
+      } else if (Email.value) /* mobile is blank and email exists */ {
+        if (!_this.checkEmail(Email.value)) {
+          e.preventDefault();
+          Mobile.parentNode.classList.add("giraffe-has-errors");
+          Email.parentNode.classList.add("giraffe-has-errors");
+          _this.addErrorToList('Valid email OR mobile opt-in is required');
+          return false;
         }
-        _this.$smsTerms.val('sms_termsandconditions');
-        _this.$smsSubscribe.val('mobilesubscribe');
-        _this.$robodialTerms.val('yes');
       }
     });
 
   }
 
-  addErrorToList(error) {
+  addErrorToList(errorText) {
     let listItem = document.createElement('li');
     listItem.classList.add("text-align-center");
-    let text = error + " is required";
-    if (error === "Email" || error === "Mobile") {
-      text = "Email OR mobile opt-in is required";
-    }
-    listItem.appendChild(document.createTextNode(text));
+    listItem.appendChild(document.createTextNode(errorText));
     this.$errors[0].appendChild(listItem);
   }
 
-}
+  checkZip(value) {
+    return (/(^\d{5}$)|(^\d{5}-\d{4}$)/).test(value);
+  }
 
+  checkEmail(value) {
+    const reg = new RegExp('[^\\.\\s@:][^\\s@:]*(?!\\.)@[^\\.\\s@]+(?:\\.[^\\.\\s@]+)*');
```

Comment:
the regex needs to start with `^` and end with `$`, or it will match email addresses anywhere in a string, like "test@test@example.com"

Reply:
ahhh, thanks.